images/macos/helpers/Xcode.Helpers.psm1

```diff
@@ -121,4 +121,29 @@ function Get-XcodePairsList {
         $result += "$watchName $phoneName"
     }
     return $result
+}
+
+function Invoke-XCVersion {
+    param(
+        [Parameter(Mandatory)]
+        [string] $Command,
+        [Parameter()]
+        [int] $RetryAttempts = 7,
+        [Parameter()]
+        [int] $PauseDuration = 1
+    )
+
+    do {
+        Write-Host "Current attempt: [$RetryAttempts]"
+        $result = Get-CommandResult -Command $Command -Multiline
+        Write-Host "Exit code: [$($result.ExitCode)]"
+        Write-Host "Output message: "
+        $($result.Output)
+        $RetryAttempts--
+        Start-Sleep -Seconds $PauseDuration
+    }
+    while(($result.ExitCode -ne 0) -and ($RetryAttempts -gt 0))
+    if ($result.ExitCode -ne 0) {
+        throw "Command [$Command] has finished with non-zero exit code. Output Message: [$($result.Output)]"
```

Comment:
I think we don't need to print output here since we have already printed it above?

Reply:
Outpup in `throw` has been removed